torch/ao/pruning/_experimental/pruner/base_structured_sparsifier.py

```diff
@@ -27,16 +131,21 @@ class BaseStructuredSparsifier(BaseSparsifier):
             be updated.
     """
 
-    def __init__(self, defaults):
+    def __init__(self, defaults, patterns=None):
         super().__init__(defaults)
+        if patterns is None:
+            patterns = _get_default_structured_pruning_patterns()
+        self.patterns = patterns
 
     def make_config_from_model(
         self,
         model: nn.Module,
-        SUPPORTED_MODULES: Set[Type] = SUPPORTED_STRUCTURED_PRUNING_MODULES,
+        SUPPORTED_MODULES: Set[Type] = None,
```

Comment:
qq: If you set it to `None`, should it be `Optional[Set]`, or is the linter ok with this?

Reply:
The linter was fine with this, but yeah it should be `Optional`